### PR TITLE
refactor: move data.hcloud_server_type(s) to Plugin Framework

### PR DESCRIPTION
**Pull request by jooola**
Related to #752

### CODE REVIEW

**Comment by jooola** [comment]
We are probably blocked by https://togithub.com/hashicorp/terraform-plugin-framework/issues/510

**Comment by jooola** [comment]
For the plugin framework upgrade test to work, we must release a patch version with https://github.com/hetznercloud/terraform-provider-hcloud/pull/984.

File: internal/servertype/data_source_test.go [hunk 1]
```diff
@@ -101,3 +144,39 @@ func TestAccDataSourceList(t *testing.T) {
...
+	all := &servertype.DDataList{}
+	all.SetRName("all")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: teste2e.PreCheck(t),
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: map[string]resource.ExternalProvider{
+					"hcloud": {
+						VersionConstraint: "1.47.0",
```
Comment (apricote) [inline internal/servertype/data_source_test.go]: Why is it okay that this tests with an older version while the single data source required patches?

**Review by apricote — APPROVED** [review]
This looks great besides the `cx11` issue and the one question I had.

File: internal/servertype/data_source_test.go (hunk 1 above)
Comment (jooola) [reply internal/servertype/data_source_test.go]: This code you reviewed is outdated.

We need the newer version that has a bug fix for the data source list, so it is only really required in the data source list test: https://github.com/hetznercloud/terraform-provider-hcloud/pull/984

Comment (apricote) [reply internal/servertype/data_source_test.go]: Oh, sorry for that. Put in that comment yesterday and focused on `internal/servertype/data_source.go` today.

**Pull request merged**